staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go

```diff
@@ -158,6 +163,12 @@ func Required(field *Path, detail string) *Error {
 	return &Error{ErrorTypeRequired, field.String(), "", detail}
 }
 
+// RequiredOneOf returns a *Error indicating "one of values required".  This is used
+// to report at least one of required values that are not provided.
+func RequiredOneOf(field *Path, fields []string, detail string) *Error {
+	return &Error{ErrorTypeRequiredOneOf, fmt.Sprintf("%s: %s", field.String(), strings.Join(fields, ",")), "", detail}
```

Comment:
Revise this to the following format?

```
&Error{ErrorTypeRequiredOneOf, field.String(), strings.Join(fields, ","), detail}
```
Then you will get error message formatted like this:

```
field: One of values is required "configMapRef,secretRef".
```

`detail` can be made optional...

Reply:
The third member of `Error` is `BadValue`. I don't think it's very meaningful to pass in `fields` as `BadValue`. I'm going to wait for more folks' thoughts :)

Reply:
fair point. My suggestion is mostly about the readability of the generated error message.

Reply:
Please *always* use field names when constructing an instance of a public struct. Why?
* It's unreadable without them: e.g. you should not need to say things like "third member of `Error` is `BadValue`"
* It makes it hard to extend the struct. Adding a member means this code no longer compiles.